core/src/test/java/org/apache/iceberg/TestSequenceNumber.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestSequenceNumber extends TableTestBase {
```

Comment:
I think it would be great to add more tests. A few possible scenarios:
- Concurrent conflicting operations
- Transactions with multiple operations
- Rollback

Reply:
Agreed, I will add more unit tests in the next commit.

Reply:
Would it also help to add a test for picked snapshots (cherry pick feature with WAP)?

Reply:
Thanks @fbocse! Let me take a look for the cherry-picked snapshot feature first.

Reply:
Since there are changes to commit workflow (MergingSnapshotProducer), it would be good to test commit-fail-retry scenarios like `TestTransaction` does.